experimental/packages/opentelemetry-instrumentation/hook.js

```diff
@@ -0,0 +1,199 @@
+/*!
+ * Copyright 2021 Datadog, Inc.
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2.0 License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+
+const specifiers = new Map();
+const isWin = process.platform === 'win32';
+
+// FIXME: Typescript extensions are added temporarily until we find a better
+// way of supporting arbitrary extensions
+const EXTENSION_RE = /\.(js|mjs|cjs|ts|mts|cts)$/;
```

Comment:
is a case insensitive regex needed here for e.g. windows?

Reply:
This is part of the vendored package. We tried to limit how much we changed it so we could more easily update if a newer version came out and we choose to bring in those changes. Also the unit tests for Windows are passing in the test suite.